README.md

```diff
@@ -1,10 +1,384 @@
 # Contrato
 
-> The official contracts implementation
+The official [contracts](#about-contracts) implementation
 
-[![Documentation](https://github.com/product-os/contrato/actions/workflows/docs.yml/badge.svg)](https://balena-io.github.io/contrato/modules/contrato.html)
+## Quickstart
 
-## Tests
+```ts
+import { Contract } from 'contrato';
+
+const osContract = new Contract({
+	type: 'sw.os',
+	slug: 'balenaos',
+	version: '6.1.2',
+	children: [
+		{ type: 'sw.service', slug: 'balena-engine', version: '20.10.43' },
+		{ type: 'sw.service', slug: 'NetworkManager', version: '0.6.0' },
+	],
+	provides: [{ type: 'sw.feature', slug: 'secureboot' }],
+});
+
+const serviceContract = new Contract({
+	type: 'sw.application',
+	slug: 'myapp',
+	requires: [
+		{ type: 'sw.service', slug: 'balena-engine', version: '>20' },
+		{ type: 'sw.feature', slug: 'secureboot' },
+	],
+});
+
+if (osContract.satisfiesChildContract(serviceContract)) {
+	console.log('myapp can be installed!');
+}
+```
+
+[![API Documentation](https://github.com/balena-io/contrato/actions/workflows/docs.yml/badge.svg)](https://balena-io.github.io/contrato/modules/contrato.html)
+
+## About contracts
+
+Contracts provide a standardized mechanism to describing _things_. A thing generally refers to something versionable, e.g. a software library, a feature, an API, etc. Relationships between things can be established via composition and referencing (`requires` and `provides`). Through this library, contracts can be validated, composed and combined.
+
+### Why build this?
+
+balena.io is a complex product with a great number of inter-conecting components. Each of the components have their own requisites, capabilities, and incompatibilities. Contracts are an effort to formally document those interfaces, and a foundation on which we can build advanced tooling to ultimately automate the process of the team, increase productivity, and remove the human element from tasks that can be performed better by a machine.
+
+The concept of contracts is generic enough that it can be applied to seemingly unrelated scenarios, from base images and OS images, to device types and backend components. Re-using the same contract "format" between them allows us to multiply the gains we get by developing complex contract-related programming modules.
```

Comment:
Is it a "format" or a "syntax"?

Reply:
Reading the definitions it does seem like syntax fits better